src/liboslexec/shadingsys.cpp

```diff
@@ -265,9 +265,26 @@ ShadingSystem::execute (ShadingContext &ctx, ShaderGroup &sas,
 
 
 const void*
-ShadingSystem::get_symbol (ShadingContext &ctx, ustring name, TypeDesc &type)
+ShadingSystem::get_symbol (ShadingContext &ctx, ustring layername,
+                           ustring symbolname, TypeDesc &type)
 {
-    return m_impl->get_symbol (ctx, name, type);
+    return m_impl->get_symbol (ctx, layername, symbolname, type);
+}
+
+
+
+const void*
+ShadingSystem::get_symbol (ShadingContext &ctx,
+                           ustring symbolname, TypeDesc &type)
+{
+    ustring layername;
+    size_t dot = symbolname.find('.');
+    if (dot != ustring::npos) {
+        // If the name contains a dot, it's intended to be layer.symbol
```

Comment:
This was really helpful in making testshade not get overly complicated. But I can certainly add a comment to the description of this function pointing out that get_symbol("layer.param") is much more expensive than get_symbol("layer", "param").